docs/standard/commandline/define-commands.md

```diff
@@ -0,0 +1,208 @@
+
+## Define aliases
+
+Both commands and options support [aliases](syntax.md#aliases). You can add an alias to an option by calling `AddAlias`:
+
+```csharp
+var option = new Option("--framework");
+option.AddAlias("-f");
+```
+
+Given this alias, the following command lines are equivalent:
+
+```console
+myapp -f net6.0
+myapp --framework net6.0
+```
+
+Command aliases work the same way.
+
+```csharp
+var command = new Command("serialize");
+command.AddAlias("serialise");
+```
+
+This code makes the following command lines equivalent:
+
+```console
+myapp serialize
+myapp serialise
+```
+
+We recommend that you minimize the number of option aliases that you define, and avoid defining certain aliases in particular. For more information, see [Short-form aliases](syntax.md#short-form-aliases).
+
+## Required options
+
+To make an option required, set its `IsRequired` property to `true`, as shown in the following example:
+
+:::code language="csharp" source="snippets/define-commands/csharp/Program2.cs" id="requiredoption" :::
+
+The options section of the command help indicates the option is required:
+
+```output
+Options:
+  --endpoint <uri> (REQUIRED)
+  --version               Show version information
+  -?, -h, --help          Show help and usage information
+```
+
+If the command line for this example app doesn't include `--endpoint`, an error message is displayed and the command handler isn't called:
+
+```output
+Option '--endpoint' is required.
+```
+
+## Hidden commands, options, and arguments
+
+To prevent a command, option, or argument from being discoverable by using tab completion or help, set its `IsHidden` property to true, as in the following example:
+
+:::code language="csharp" source="snippets/define-commands/csharp/Program2.cs" id="hiddenoption" :::
+
+The options section of this example's command help omits the `--endpoint` option.
+
+```output
+Options:
+  --version               Show version information
+  -?, -h, --help          Show help and usage information
+```
+
+## Multiple arguments
+
+By default, you can repeat the option name to specify multiple arguments for an option that has maximum [arity](syntax.md#argument-arity) greater than one.
+
+```console
+myapp --items one --items two --items three
+```
+
+To allow multiple arguments without repeating the option name, set `Option.AllowMultipleArgumentsPerToken` to `true`. This setting lets you enter the following command line.
+
+```console
+myapp --items one two three
+```
+
+The same setting has a different effect if maximum argument arity is 1. It allows you to repeat an option but takes only the last value on the line. In the following example, the value `three` would be passed to the app.
+
+```console
+myapp --item one --item two --item three
+```
+
+## List valid argument values
```

Comment:
Shall we add something along the lines of: 

If your list of values is contained in an enum, you can just use the enum as the type of the option to get similar support.